src/metabase/pulse/render/table.clj

```diff
@@ -74,12 +74,18 @@
   [score]
   (int (* (/ score 100.0) max-bar-width)))
 
+(defn- truncate-text [text]
+  (let [max-column-character-limit 16]
+    (if (> (count text) max-column-character-limit)
+      (str (subs text 0 max-column-character-limit) "...")
+      text)))
+
 (defn- render-table-head [{:keys [bar-width row]}]
   [:thead
    [:tr
     (for [header-cell row]
-      [:th {:style (style/style (row-style-for-type header-cell) (heading-style-for-type header-cell) {:min-width :42px})}
-       (h header-cell)])
+      [:th {:style (style/style (row-style-for-type header-cell) (heading-style-for-type header-cell) {:min-width :42px}) :title header-cell}
+       (h (truncate-text (str header-cell)))])
```

Comment:
I can't figure out how to get the text out of `header-cell` without wrapping it in a `str` function

Reply:
Thanks for the suggestion 👍 I did what you suggested, and it looks a lot simpler.